Complete implementation with 100% coverage

nitika-goel opened the pull request:
Summary of changes made in contract:
- changed balanceOf function to keep in line with stndard(locked tokens are ignored in balanceOf)
- changed transfer and transferFrom to standard implementations
- added return type to lock function to maintain standard
- changes in data mapping and variable naming

Other changes
- added test cases
- added travis

nitika-goel committed:
Add travis settings

nitika-goel committed:
Travis with code coverage

nitika-goel committed:
Changes to standardize ERC20 functions

nitika-goel committed:
Updated test cases

nitika-goel committed:
Changes to lockReason

nitika-goel committed:
Updated test-cases

nitika-goel committed:
Added checks

nitika-goel committed:
Added return type to functions

nitika-goel committed:
Fixed testing bugs

nitika-goel committed:
Adding test case for transfer

nitika-goel committed:
Removed unused variables

davesag commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+dist: trusty
+language: node_js
+node_js:
+  - '7'
```
`node 8.11.3` is the current Long Term Supported release so I recommend testing against that.

davesag commented on .travis.yml:
```diff
@@ -0,0 +1,16 @@
+  - '7'
+install:
+  - npm install -g truffle
+  - npm install -g ganache-cli
```
not sure you need this as truffle actually includes ganache.

davesag commented on contracts/LockableToken.sol:
```diff
@@ -2,19 +2,16 @@ pragma solidity ^0.4.24;
 
     /**
-     * @dev Utilities for which a token can be locked
+     * @dev Utilities for which a user's token have been locked
```
Maybe change the wording to "Reasons why a user's tokens have been locked"

davesag commented on contracts/LockableToken.sol:
```diff
@@ -49,68 +46,118 @@ contract LockableToken {
+    
+    /**
+     * @dev Locks a specified amount of tokens against an address,
+     *      for a specified purpose and time
```
change 'purpose' to 'reason' to be consistent

davesag commented on contracts/LockableToken.sol:
```diff
@@ -49,68 +46,118 @@ contract LockableToken {
+    /**
+     * @dev Locks a specified amount of tokens against an address,
+     *      for a specified purpose and time
+     * @param _reason The purpose to lock tokens
```
change to "The reason the tokens are locked"

davesag commented on contracts/LockableToken.sol:
```diff
@@ -49,68 +46,118 @@ contract LockableToken {
+        public
+        returns (bool)
+    {
+        uint256 validUntil=block.timestamp.add(_time);
```
space before and after `=`

davesag commented on contracts/LockableToken.sol:
```diff
@@ -49,68 +46,118 @@ contract LockableToken {
 
     /**
      * @dev Returns tokens locked for a specified address for a
      *      specified purpose at a specified time
```
change 'purpose' to 'reason' to be consistent

davesag commented on test/LockableToken-creation_test.js:
```diff
@@ -1,14 +1,18 @@
   const supply = 1000
-  const lockedFor = 'CA'
-  const lockedAmount = 500
+  const lockReason =
```
In JS these can actually just be strings and `web3` will handle the conversion to `bytes32`.  so you could say
```
const lockReason1 = 'some reason'
const lockReason2 = 'some other reason'
```

davesag commented on test/LockableToken-creation_test.js:
```diff
@@ -27,21 +31,153 @@ contract('LockableToken', ([owner]) => {
+
+    it('cannot transfer tokens to null address', async function() {
+      await assertRevert(
+        token.transfer(0x0000000000000000000000000000000000000000, 100, {
```
make `0x0000000000000000000000000000000000000000` a constant.

davesag reviewed (changes requested):
Cool.  Just a few wording and stylistic changes requested.

nitika-goel committed:
PR Review CHanges

nitika-goel committed:
Updated node version

nitika-goel committed:
Updated readme

nitika-goel committed:
Changes in comments

nitika-goel commented:
@davesag , thanks for reviewing. Have made a few changes in comments, updated readme and incorporated the changes highlighted. Let me know what you think.
